src/ddl/v4_schema.sql

```diff
@@ -15,77 +15,63 @@ GRANT ALL ON covid.* TO 'user';
 -- --------------------------------
 
 CREATE TABLE geo_dim (
-    `geo_key_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
-    `geo_type` VARCHAR(12),
-    `geo_value` VARCHAR(12),
-    `compressed_geo_key` VARCHAR(100),
-    
-    PRIMARY KEY (`geo_key_id`) USING BTREE,
-    UNIQUE INDEX `compressed_geo_key_ind` (`compressed_geo_key`) USING BTREE
-);
+    `geo_key_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `geo_type` VARCHAR(12) NOT NULL,
+    `geo_value` VARCHAR(12) NOT NULL,
+
+    UNIQUE INDEX `geo_dim_index` (`geo_type`, `geo_value`)
+) ENGINE=InnoDB;
 
 
 CREATE TABLE signal_dim (
-    `signal_key_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT, 
-    `source` VARCHAR(32),
-    `signal` VARCHAR(64),
-    `compressed_signal_key` VARCHAR(100),
-    
-    PRIMARY KEY (`signal_key_id`) USING BTREE,
-    UNIQUE INDEX `compressed_signal_key_ind` (`compressed_signal_key`) USING BTREE
+    `signal_key_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `source` VARCHAR(32) NOT NULL,
+    `signal` VARCHAR(64) NOT NULL,
+
+    UNIQUE INDEX `signal_dim_index` (`source`, `signal`)
 ) ENGINE=InnoDB;
 
+CREATE TABLE strat_dim (
+    `strat_key_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `stratification_name` VARCHAR(64) NOT NULL UNIQUE,
+    `stratification_descr` VARCHAR(64) NOT NULL
+) ENGINE=InnoDB;
+INSERT INTO strat_dim VALUES (1, 'NO_STRATIFICATION', '');
 
 CREATE TABLE signal_history (
-    `signal_data_id` BIGINT(20) UNSIGNED NOT NULL,
-    `signal_key_id` BIGINT(20) UNSIGNED,
-    `geo_key_id` BIGINT(20) UNSIGNED,
-    `demog_key_id` BIGINT(20) UNSIGNED,  -- TODO: for future use ; also rename s/demog/stratification/
-    `issue` INT(11),
+    `signal_data_id` BIGINT(20) UNSIGNED NOT NULL PRIMARY KEY,
+    `signal_key_id` BIGINT(20) UNSIGNED NOT NULL,
+    `geo_key_id` BIGINT(20) UNSIGNED NOT NULL,
+    `strat_key_id` BIGINT(20) UNSIGNED NOT NULL DEFAULT 1,  -- TODO: for future use
+    `issue` INT(11) NOT NULL,
     `data_as_of_dt` DATETIME(0),  -- TODO: for future use ; also "as_of" is problematic and should be renamed
```

Comment:
rename before migrating the schema?

Reply:
Renaming a column is cheap and easy, and it isnt a priority that we do this right now.  It is important that we have the "column space" reserved with the appropriate datatype because adding it later isnt so simple.  TBQH i would like to rename several of the columns (and we have talked about it before) but the hope was that we finish the real work first and deal with the naming later.